### ISSUE TITLE
Fast sync crashes with "ERROR:: fast_sync() takes exactly 3 arguments (2 given)"

### ISSUE THREAD

**Issue by hoffleit** [MediaBrowser/plugin.video.emby#321]
Since the commit ae34b03aa7801adc744ebe5ff0244c3e2f94630f ("added progress output when receiving changes from Emby server"), I'm getting this error:

```
EMBY.library -> ERROR:: fast_sync() takes exactly 3 arguments (2 given)
  Traceback (most recent call last):
    File "/home/gregor/.var/app/tv.kodi.Kodi/data/addons/plugin.video.emby/resources/lib/library.py", line 493, in startup
      self.get_fast_sync()
    File "/home/gregor/.var/app/tv.kodi.Kodi/data/addons/plugin.video.emby/resources/lib/library.py", line 542, in get_fast_sync
      self.fast_sync(enable_fast_sync)
    File "/home/gregor/.var/app/tv.kodi.Kodi/data/addons/plugin.video.emby/resources/lib/helper/wrapper.py", line 39, in wrapper
      result = func(self, dialog=dialog, *args, **kwargs)
 TypeError: fast_sync() takes exactly 3 arguments (2 given)
```

Both systems I tried show this behavior (CoreElec 9.2.0 (Kodi 18.4) with Python 2.7.16, as well as the current flatpak 18.4-Leia from https://flathub.org/apps/details/tv.kodi.Kodi on a amd64 Debian GNU/Linux 10).

Changing the function declaration of the decorator @progress() makes this error go away for me on both systems:

```
diff --git a/resources/lib/helper/wrapper.py b/resources/lib/helper/wrapper.py
index bf2f87ae..a5dca314 100644
--- a/resources/lib/helper/wrapper.py
+++ b/resources/lib/helper/wrapper.py
@@ -20,10 +20,15 @@ def progress(message=None):
     ''' Will start and close the progress dialog.
     '''
     def decorator(func):
-        def wrapper(self, item=None, *args, **kwargs):
+        def wrapper(self, *args, **kwargs):
 
             dialog = xbmcgui.DialogProgressBG()
 
+            if "item" in kwargs:
+                item = kwargs["item"]
+            else:
+                item = None
+
             if item and type(item) == dict:
 
                 dialog.create(_('addon_name'), "%s %s" % (_('gathering'), item['Name']))
@@ -32,9 +37,6 @@ def progress(message=None):
                 dialog.create(_('addon_name'), message)
                 LOG.info("Processing %s", message)
 
-            if item:
-                args = (item,) + args
-
             try:
                 result = func(self, dialog=dialog, *args, **kwargs)
                 dialog.close()
```

Reading this: https://stackoverflow.com/questions/9872824/calling-a-python-function-with-args-kwargs-and-optional-default-arguments, I'm a little bit afraid that the current declaration `def wrapper(self, item=None, *args, **kwargs)` hits a blind spot, where the behavior is not deterministic.

1. The CoreElec box: 

>     Starting Kodi (18.4 Git:e95dc442265375aa0cd75284eeaf70ad99b04862). Platform: Linux ARM 32-bit
>     Using Release Kodi x32 build
>     Kodi compiled 2019-09-15 by GCC 8.3.0 for Linux ARM 32-bit version 3.14.29 (200221)
>     Running on CoreELEC (official): 9.2.0, kernel: Linux ARM 64-bit version 3.14.29 aarch64
>     FFmpeg version/source: 4.0.4-Kodi
>     Host CPU: AArch64 Processor rev 4 (aarch64), 4 cores available
>     ARM Features: Neon enabled
>     ...
>     EMBY.entrypoint.service -> Python Version: 2.7.16 (default, Sep 15 2019, 17:03:58) [GCC 8.3.0]
>     ...

2. The Flatpak on amd64 Debian 10:

>     Starting Kodi (18.4 Git:20190906-nogitfound). Platform: Linux x86 64-bit
>     Using Release Kodi x64 build
>     Kodi compiled 2019-09-06 by GCC 8.3.0 for Linux x86 64-bit version 4.19.26 (267034)
>     Running on Freedesktop.org 18.08.38 (Flatpak runtime), kernel: Linux x86 64-bit version 4.19.0-6-amd64
>     NOTICE: FFmpeg version/source: 4.0.4-Kodi
>     ...
>     EMBY.entrypoint.service -> Python Version: 2.7.16 (default, Sep  6 2019, 14:24:22) [GCC 8.3.0]
>     ...

 

Emby for Kodi was v4.1.15 on both systems

**Comment by angelblue05** [comment]
Can you try a simple one line change?

In the library.py file, line 542
self.fast_sync(plugin=enable_fast_sync)

Does that correct it?

**Comment by angelblue05** [comment]
I mean try this, without your changes. Thanks.

**Comment by hoffleit** [comment]
That did it. Rolled back my changes and applied that change and the problem is gone. Thanks!

**Comment by angelblue05** [comment]
Thanks for confirming, I’ll release a new build with that fix in a bit.

**Comment by angelblue05** [comment]
Fixed with 4.1.16
